homeassistant/components/media_player/kodi.py

```diff
@@ -262,6 +296,17 @@ def on_hass_stop(event):
         else:
             self._ws_server = None
 
+        # Script creation for the turn on/off config options
+        if turn_on_action is not None:
+            turn_on_action = script.Script(
+                self.hass, turn_on_action,
+                "Kodi Turn ON action script",
+                partial(self.async_update_ha_state, force_refresh=True))
+        if turn_off_action is not None:
+            turn_off_action = script.Script(
+                self.hass, _check_deprecated_turn_off(turn_off_action),
+                "Kodi Turn OFF action script", self.async_on_quit)
```

Comment:
Don't include `async_on_quit` here. We want to issue the command and rely on Kodi to report the change instead of assuming it happens.